### DISCUSSION TITLE
Skills and training

### DISCUSSION THREAD

**Discussion by longr** [NERC-CEH/rse_group#51] [General]
I wanted to start a discussion on skills and training we need to deliver, and what might come under that.   I think the best way it to make a comment for each area and we can use the "replies" to discuss details perhaps?

**Comment by longr** [comment]
Introduction to Git

**Reply by longr** [reply]
@matthewcoole has started this with a fork of the Carpentries Git lesson ([https://github.com/ukceh-rse/git-novice/](https://github.com/ukceh-rse/git-novice/)).    We probably need to add some more episodes on basic cli navigation so that users know how to navigate to their files.

Perhaps add more supplementary material on git from VSCode, and RStudio and so on.

**Comment by longr** [comment]
Github and Sharing

**Reply by longr** [reply]
What do we need in a GitHub course?
- Forking
- Branches
- Pull Request
- Issues
- Discussion
- ?

**Reply by metazool** [reply]
* (Slightly subjective) material on "what makes a healthy project" with some positive examples (ideally UKCEH ones - finding them would be part of a curation exercise!)
* Actions (in overview, with detail on testing in CI in an intermediate course)
* Projects, for just-enough Kanban?

**Reply by longr** [reply]
Good suggestions. We definitely need to cover actions and health project. Git course touches on license and citation, but README and badges is needed I think.

I think it is nice if we can pull these together into a story, makes it clearer why you might use each tool, but definitely need to teach these in way.

**Reply by mattjbr123** [reply]
This caught my interest in a pangeo meeting lately: https://scipy2024-githubactionstutorial.readthedocs.io/en/latest/batch-computing.html
Probably (haven't looked at it in detail) too advanced for our needs at this stage, I just enjoyed the hacky idea that you could use GH action as a sortof batch compute HPC-like thing tbh

**Comment by longr** [comment]
Databases - SQL and SQLite.

**Reply by longr** [reply]
I have found myself not knowing much about databases at all, and needing this in a current project.  I have touched on creating an SQL and SQLite db, but nothing more.  After conversations with @matthewcoole I realised there is a lot more to this including database normalisation (https://www.datacamp.com/tutorial/normalization-in-sql) which people may need to know about.  Perhaps we could look up carpentries style or other courses on databases and tweak to CEH needs?   SQL vs SQlite, accessing both from Python and R, using datalabs/ internal servers to host databases, and even APIs to access them?

**Comment by longr** [comment]
GitHub for project managers / PI's / product owners and so on

**Reply by longr** [reply]
Should we do a course more focused on Project management, issues and not coding for those who will be leading projects and doing less of the coding?
